Documentation Rewrite (#417)
stoplightio/spectral: docs/guides/workflows.md

# Workflow

When and where should you use Spectral? 

> It depends a lot how you are creating and managing your API description documents, but probably wherever they are being made. [Stoplight Studio](https://stoplight.io/spectral/) users have Spectral baked right into the editor, but if you are using another text-editor or IDE you can switch to the command-line and run Spectral [as a CLI tool][cli].

Reviewer: ```suggestion
It depends a lot how you are creating and managing your API description documents, but probably wherever they are being made. [Stoplight Studio](https://stoplight.io/studio/) users have Spectral baked right into the editor, but if you are using another text-editor or IDE you can switch to the command-line and run Spectral [as a CLI tool][cli].
```

## Linting Design-First Workflows

> If the developer is just in the early stages of planning and designing the API, they could run Spectral against their design docs and get feedback very early on. If they are using Studio, Spectral will be running automatically as they work, without the developer even needing to switch to the CLI.

Reviewer: Would using bullet points make sense here?
```
suggestion
- Run Spectral CLI against design docs and get feedback very early on.
- Run Spectral in Studio automatically as you work, without the needing to switch to the CLI.
```

> Seeing these errors and warnings will help nudge the developer towards creating consistent APIs, quickly and easily, without needing to have "OpenAPI Gatekeepers" to enforce the rules manually. Those folks are not infallible, they can miss things, but Spectral can be used to free those people up for bigger and better things.

Reviewer: ```suggestion
- spot errors and warning early
- create consistent APIs, quickly and easily
- ditch "OpenAPI Gatekeepers" to enforce the rules manually

Those folks are not infallible, they can miss things, but Spectral can be used to free those people up for bigger and better things.
```

Author: I dunno I think I like the flow of the sentence as it is? I changed the other wordy sentence to use bullet points though :D

Reviewer: I guess I'm more of a Haiku person and you're more of a prose person :) Looking great!

> ## Linting Code-First Workflows

Reviewer: Nice, great write up! I'm thinking that maybe we could remove it from the docs and leave it for a short blog post on SL page? What do you think? Does blog or docs make for a better place for this type of content?

Author: Blogs are temporary and ephemeral but docs are lasting. This is touching on opinions but still needs to be readily available for people as they need to know when and how to use spectral.
